### ISSUE TITLE
Refactor: Use the capabilities to defined if an Agent is upgradable or not

### ISSUE THREAD

**Issue by ph** [elastic/beats#24908]
Today we can use https://github.com/elastic/beats/blob/64f70785c0911eeb6f3f6ce5264f61544844ca0f/x-pack/elastic-agent/pkg/agent/application/upgrade/upgrade.go#L78 to define if a release is upgradable or not. We have added the concept of **capabilties** in https://github.com/elastic/beats/pull/23848 

We should if to the **capabitilies** files, it could look like this:

```
capabilities:
-  upgrade: false
```

@ruflin  @mostlyjason WDYT if make it generic on the action that we support https://github.com/elastic/beats/tree/1f1fae56057dce0604f72f2cf0099f9a6f2b75aa/x-pack/elastic-agent/pkg/agent/application/pipeline/actions/handlers? 

```
capabilities:
- rule: deny
  action: Upgrade
```

**Comment by ruflin** [comment]
++ on adding it to capabilities. 

@ph Can you elaborate on the action approach? You mean it would cover multiple cases at once directly? If yes, kind of like the idea.

We will need to make sure independent of the implementation, that an Agent reports its capabilities so Fleet can disable certain UI components / API calls.

**Comment by ph** [comment]
@ruflin Yeah, This is my point, we could cover multiple cases with a single strategy. Every comment send from Fleet is an action, including upgrade so if we add something to control which kind of action we can executed on the agent we actually support the current use case of an upgrade an any potential action we would like to restrict.

And by doing so we only need to figure out "how capabilities" are surfaced back in the UI.
